src/awkward/_v2/contents/recordarray.py

```diff
@@ -585,10 +572,9 @@ def mergemany(self, others):
         for_each_field = []
         for field in self.contents:
             trimmed = field[0 : self.length]
-            for_each_field.append([field])
+            for_each_field.append([trimmed])
```

Comment:
So we made the `trimmed` and simply forgot to pass that on instead of `field`?

Good catch!

Reply:
Yes. I'm assuming that `trimmed` was in there for good reason, so I've modified the succeeding line.